packages/react-label/Spec.md

```diff
@@ -6,58 +6,175 @@ _Description and use cases of this component_
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+The Label component should be simple as shown below. It will just need the text to be rendered. For the required label, it has the option of being a shorthand slot that will allow to customize the label's required text.
+
+```tsx
+<Label>Label</Label>
+
+<Label disabled>Label</Label>
+
+<Label required>Label</Label>
+
+<Label required="**">
+  Label
+</Label>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+- A Label can be rendered with an asterisk or custom text when is set as `required`.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+```ts
+/**
+ * Label Props
+ * {@docCategory Label}
+ */
+export interface LabelProps extends ComponentProps, React.LabelHTMLAttributes<HTMLElement> {
+  /**
+   * Renders the label as disabled
+   * @defaultvalue false
+   */
+  disabled?: boolean;
+
+  /**
+   * Whether the associated form field is required or not. If true it will be an asterisk, otherwise it will be what is provided.
+   * @defaultvalue false
+   */
+  required?: boolean | ShorthandProps<ComponentProps>;
+
+  /**
+   * A label supports different font sizes, see tokens for reference.
+   * @defaultvalue 'medium'
+   */
+  size?: 'small' | 'medium' | 'large';
+
+  /**
+   * A label supports semibold/strong fontweight. When size is set to large, label strong will be set by default.
+   * @defaultvalue false
+   */
+  strong?: boolean;
+}
+
+/**
+ * Names of the shorthand properties in LabelProps
+ * {@docCategory Label}
+ */
+export type LabelShorthandProps = 'required';
+
+/**
+ * Names of LabelProps that have a default value in useLabel
+ * {@docCategory Label}
+ */
+export type LabelDefaultedProps = never;
+
+/**
+ * State used in rendering Label
+ * {@docCategory Label}
+ */
+export interface LabelState extends ComponentState<LabelProps, LabelShorthandProps, LabelDefaultedProps> {
+  /**
+   * Ref to the root element
+   */
+  ref: React.Ref<HTMLElement>;
+}
+```
 
 ## Structure
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+### Public
 
-## Migration
+```tsx
+<Label required>
+  I'm a Label
+</Label>
+```
+
+### DOM
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+```tsx
+<label {/*Label*/} class="...">
+  I'm a Label
+  <span {/*required*/} class="...">*</span>
+</label>
+```
 
-- _Migration from v8_
-- _Migration from v0_
+### Internal
+
+```tsx
+<slots.root {...slotProps.root}>
+  {state.children}
+  <slots.required {...slotProps.required} />
+</slots.root>
+```
+
+## Migration
+
+See MIGRATION.md
```

Comment:
You can preemptively add a link here so that it correctly links once this merges:

```suggestion
See [MIGRATION.md](https://github.com/microsoft/fluentui/tree/master/packages/react-label/MIGRATION.md)
```